components/selection/selection-header.js

```diff
@@ -0,0 +1,188 @@
+import '../overflow-group/overflow-group.js';
+import './selection-select-all.js';
+import './selection-select-all-pages.js';
+import './selection-summary.js';
+import { css, html, LitElement } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+import { SelectionObserverMixin } from './selection-observer-mixin.js';
+
+/**
+ * A header for selection components (e.g. list, table-wrapper) containing select-all, etc.
+ * @slot - Responsive container using `d2l-overflow-group` for `d2l-selection-action` elements
+ */
+export class SelectionHeader extends SelectionObserverMixin(RtlMixin(LocalizeCoreElement(LitElement))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether to render select-all and selection summary
+			 * @type {boolean}
+			 */
+			noSelection: { type: Boolean, attribute: 'no-selection' },
+			/**
+			 * Disables sticky positioning for the header
+			 * @type {boolean}
+			 */
+			noSticky: { type: Boolean, attribute: 'no-sticky' },
+			/**
+			 * Whether all pages can be selected
+			 * @type {boolean}
+			 */
+			selectAllPagesAllowed: { type: Boolean, attribute: 'select-all-pages-allowed' },
+			_hasActions: { state: true },
+			_scrolled: { type: Boolean, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				background-color: var(--d2l-selection-header-background-color, white);
+				display: block;
+				position: sticky;
+				top: 0;
+			}
+			:host([no-sticky]) {
+				background-color: transparent;
+				position: static;
+			}
+			.d2l-selection-header-shadow {
+				transition: box-shadow 200ms ease-out;
```

Comment:
I notice that we previously did not have a media query to disable the transition. It is super minor, but if the consumer prefers reduced motion we should just avoid it entirely unless necessary (like a spinner), plus it can help avoid issues with visual-diffs. We used to write them the opposite way (using `prefers-reduced-motion: reduce`), but now that we no longer support legacy Edge, we can write the animation as a progressive enhancement. I think we should include it:
```css
@media (prefers-reduced-motion: no-preference) {
	.d2l-selection-header-shadow {
		transition: box-shadow 200ms ease-out;
	}
}
```